File: client-clojure/src/sirqul_io_t_platform/specs/cell_carrier.clj
(ns sirqul-io-t-platform.specs.cell-carrier
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def cell-carrier-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :uid) string?
   (ds/opt :display) string?
   (ds/opt :emailFormat) string?
   (ds/opt :apiUrl) string?
   (ds/opt :authUrl) string?
   (ds/opt :shortCode) string?
   (ds/opt :clientId) string?
   (ds/opt :clientSecret) string?
   })

(def cell-carrier-spec
  (ds/spec
    {:name ::cell-carrier
     :spec cell-carrier-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/CellCarrier.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CellCarrier._

case class CellCarrier (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
uid: Option[String],
display: Option[String],
emailFormat: Option[String],
apiUrl: Option[String],
authUrl: Option[String],
shortCode: Option[String],
clientId: Option[String],
clientSecret: Option[String])

object CellCarrier {
  import DateTimeCodecs._

  implicit val CellCarrierCodecJson: CodecJson[CellCarrier] = CodecJson.derive[CellCarrier]
  implicit val CellCarrierDecoder: EntityDecoder[CellCarrier] = jsonOf[CellCarrier]
  implicit val CellCarrierEncoder: EntityEncoder[CellCarrier] = jsonEncoderOf[CellCarrier]
}
